src/pages/Home.tsx

```diff
@@ -1,40 +1,55 @@
 import { useEffect, useState } from "react";
 import { api } from "../api/api";
-import { useNavigate } from 'react-router-dom';
+import { useNavigate } from "react-router-dom";
 import { Header } from "../components/Header";
 import { Button } from "react-bootstrap";
 import { useTranslation } from "react-i18next";
+import store from "../store/RootStore";
 
 const Home = () => {
   const { t } = useTranslation();
   const navigate = useNavigate();
-  const [message, setMessage] = useState<string>(t('loading'));
+  const [message, setMessage] = useState<string>(t("loading"));
 
   useEffect(() => {
-    handleGetMessage();
+    if (!store.user.profile?.selected_profile) {
+      navigate("/profile-selection");
+    } else {
+      handleGetMessage();
+    }
   }, []);
```

Comment:
Tutaj ta sama sytuacja co wyżej z tym eslintem, dodałbym:
```suggestion
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```